clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -75,6 +74,7 @@
     private final int autoCommitIntervalMs;
     private final ConsumerInterceptors<?, ?> interceptors;
     private final boolean excludeInternalTopics;
+    private final AtomicInteger pendingAsyncCommits;
```

Comment:
Thinking about this a little more, I'm not sure adding coordinator discovery to this loop helps us. The issue is that a coordinator disconnect will cause all of the pending offset commit requests to fail. So I think the logic should be something like:
1. Check if the coordinator exists. If not, close immediately.
2. Check if there are pending requests. An optimization here is to check the rebalance state and exit immediately if we are rebalancing (after sending LeaveGroup).
3. Wait until all pending requests have returned. If the coordinator disconnects at any time, exit immediately.
Does that make sense or not?